### ISSUE TITLE
failed to reproduce DSNAS paper performance

### ISSUE THREAD

**Issue by khy0809** [SNAS-Series/SNAS-Series#2]
Dear author, 
thank you for this great work and code. 
I tried to reproduce the performance of DSNAS by following the instructions on README (firstly search supernet for 80 epochs, and then continue searching with train_imagenet_child.py)
but failed to reproduce the results. As shown in the attached curve below, I got only 73.57 top-1 accuracy, which is much lower than 74.4 as the paper claims. 
![image](https://user-images.githubusercontent.com/36264092/79085463-a0296580-7d73-11ea-8822-7644a76f86f6.png)

**Comment by skhu101** [comment]
Very thanks for running our codes. It seems that the evaluation curve accuracy is a little bit strange. Could you send me your generated log files and script folder to my email (skhu [at] se [dot] cuhk [dot] edu [dot] hk)? I will have a check and give you the response.

**Comment by khy0809** [comment]
Since "loss_alpha" is a loss value, it should be in [0, inf]. However, according to the current implementation it is in [-inf, 0], so the sign should be changed. Is this a simple bug, or am I misunderstanding something?
https://github.com/SNAS-Series/SNAS-Series/blob/568102e527e934d21a8b4eec7ee8b1e0f1204f52/DSNAS/network.py#L119
https://github.com/SNAS-Series/SNAS-Series/blob/568102e527e934d21a8b4eec7ee8b1e0f1204f52/DSNAS/network.py#L141

**Comment by skhu101** [comment]
No, it's not a bug. Here we use loss_alpha to calculate the gradient of the first part in the following equation.
<img width="534" alt="Screenshot 2020-04-14 at 7 55 07 PM" src="https://user-images.githubusercontent.com/62331757/79222293-d9ec9000-7e89-11ea-8371-19ca8b293a75.png">

**Comment by skhu101** [comment]
Thanks for your interest in our paper. We run the experiments twice in the second searching process by using the checkpoint (model.pth-80) you sent us  and get the results (74.4 and 74.2). The corresponding tensorboard log file is in https://drive.google.com/drive/folders/1DNh55QVCuOFLJNBkvXcLpxBRukCI7ujU?usp=sharing. It's possible that dropout layer makes the variance larger, as we have also found a very small variance with the absense of dropout (74.2+-0.1). Maybe you can also run the experiments several times to check the result with/without dropout layer. If you have any further questions, welcome to discuss with us.
<img width="1044" alt="Screenshot 2020-04-19 at 11 02 28 AM" src="https://user-images.githubusercontent.com/62331757/79678359-44f0ea80-822d-11ea-8f93-114cc9624d69.png">

**Comment by akhauriyash** [comment]
Hello @skhu101 
Hope you are doing well.
I am doing some research and have used your library (which is excellent :) ) as a baseline.
I have managed to replicate your accuracy (74.2, not 74.4). However, my training accuracy is always much higher when compared to the training accuracy curve your tensorboard has. Should I be worried? Maybe I am leaving some performance on the table. Note that in the image I have uploaded, 'ours' refers to my replication of the 'ref_sim' which is the one you have uploaded. 'ours' uses a different model with 309 MFLOPs.

![image](https://user-images.githubusercontent.com/19608816/109757056-0b3f8d00-7c0f-11eb-9876-dc27d76c2cb9.png)


I also get a lot of these errors (maybe 14 per epoch) during the training. Maybe this is causing skipping of entire mini batches? 

```
/opt/conda/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:754: UserWarning: Possibly corrupt EXIF data.  Expecting to read 2555904 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
```
Do let me know your thoughts! Thank you!
